drivers/console/aarch64/console.S

```diff
@@ -95,10 +95,15 @@ endfunc console_getc
 	 * data that hasn't been output. It returns 0
 	 * upon successful completion, otherwise it
 	 * returns -1.
-	 * Clobber list : x0, x1
+	 * Clobber list : x0, x1, x2, x3, x4
 	 * ---------------------------------------------
 	 */
 func console_flush
+#if COREBOOT
+	mov	x4, x30
+	bl	coreboot_cbmc_flush
+	mov	x30, x4
+#endif
```

Comment:
> Yes, that was what I was envisaging. The "splitting" console API would be a struct of function pointers corresponding to the existing console API. This would be much like the PSCI pm-ops API. Each console driver would register a struct containing function pointers unique to that driver. The platform would also provide a macro to determine the number of supported consoles (i.e. how much memory to allocate for the array of structs).

Sorry, I'm still not quite sure I understand you right. Do you want me to change the whole API in the generic code (meaning that console_putc would no longer just call console_core_putc, but would instead look up a function pointer to call from a struct that was previously registered)? I don't suppose you do, because it sounds like you want to keep all of these changes out of the generic code?

My problem right now is that the generic code is just calling console_core_putc, and the ti/16550_console.S driver that I want to continue to use unchanged is implementing console_core_putc. If I want to write a splitting driver that can call several implementations, it also needs to implement console_core_putc... but then that would clash with the existing implementation from the 16550 driver. As long as the generic API just consists of calling well-known function names, how could I implement a splitting driver that can wrap the unchanged existing console drivers without naming clashes?